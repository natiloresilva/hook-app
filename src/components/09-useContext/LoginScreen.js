import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {

    const { setUser, user } = useContext(UserContext)

    return (
        <>
            <h1>Login Screen</h1>
            <hr />

            <pre>
                {JSON.stringify(user, null, 3)}
            </pre>

            <button
                className='btn btn-primary'
                onClick={() => setUser({
                    id: 123,
                    name: 'Galileo',
                    email: 'gali@gmail.com'
                })}
            >
                Login
            </button>
        </>
    );
};
