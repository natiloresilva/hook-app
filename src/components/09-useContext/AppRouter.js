import React from "react";
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
} from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />

                <div className='container'>
                    <Switch>
                        <Route path="/about" component={AboutScreen} />
                        <Route path="/login" component={LoginScreen} />
                        <Route path="/" component={HomeScreen} />{" "}
                        {/* Va al final porque sino fallarán las rutas, otra solución es poner <Route exact path='/' component={HomeScreen} /> */}
                        {/*<Route component={HomeScreen} />  Si no se encuentra el path, siempre nos dirigirá a Home /> .También podemos usar el Redirect */}
                        <Redirect to="/" />
                    </Switch>
                </div>
            </div>
        </Router>
    );
};
