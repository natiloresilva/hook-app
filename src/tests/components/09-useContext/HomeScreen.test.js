import React from 'react';
import { mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

//Config para utilizar el mount, ya que hay problemas de compatibilidad con la versi√≥n 17 de React
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

Enzyme.configure({ adapter: new Adapter() })

describe('Pruebas en <HomeScreen />', () => {

    const user = {
        name: 'Fernando',
        email: 'fernando@gmail.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    );

    test('debe de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    })


})
