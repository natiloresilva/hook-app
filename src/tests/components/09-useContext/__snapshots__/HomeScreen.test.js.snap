// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> debe de mostrarse correctamente 1`] = `
Array [
  <h1>
    Home Screen
  </h1>,
  <hr />,
  <pre>
    {
   "name": "Fernando",
   "email": "fernando@gmail.com"
}
  </pre>,
]
`;
